workspace/scm.py

```diff
@@ -193,6 +193,15 @@ def merge_branch(branch, squash=False, strategy=None):
         cmd.append('--squash')
     if strategy:
         cmd.append('--strategy=' + strategy)
+
+    current = current_branch()  # uses current_branch() method  to get current branch
+    message = ""
+    if current:
```

Comment:
in what situation would current_branch() return nothing? If that happens it seems like we have an issue and I am not sure we want to continue because then the merge message will be empty.

Reply:
For merge to work, it is safe to assume there is always a current branch. I thought it might not work if checkout is done for a commit, but it also works for that situation as well. I don't remember what situation would return None, so it's ok to not check for current.